frame/support/src/traits/tokens/fungible/balanced.rs

```diff
@@ -0,0 +1,359 @@
+	///
+	/// If `who` doesn't exist, nothing is done and an `Err` returned. This could happen because it
+	/// the account doesn't yet exist and it isn't possible to create it under the current
+	/// circumstances and with `value` in it.
+	fn deposit(
+		who: &AccountId,
+		value: Self::Balance,
+	) -> Result<DebtOf<AccountId, Self>, DispatchError>;
+
+	/// Removes `value` balance from `who` account if possible.
+	///
+	/// If the removal is not possible, then it returns `Err` and nothing is changed.
+	///
+	/// If the operation is successful, this will return `Ok` with a `NegativeImbalance` whose value
+	/// is no less than `value`. It may be more in the case that removing it reduced it below
+	/// `Self::minimum_balance()`.
+	fn withdraw(
+		who: &AccountId,
+		value: Self::Balance,
+		//TODO: liveness: ExistenceRequirement,
+	) -> Result<CreditOf<AccountId, Self>, DispatchError>;
+
+	/// The balance of `who` is increased in order to counter `credit`. If the whole of `credit`
+	/// cannot be countered, then nothing is changed and the original `credit` is returned in an
+	/// `Err`.
+	///
+	/// Please note: If `credit.peek()` is less than `Self::minimum_balance()`, then `who` must
+	/// already exist for this to succeed.
+	fn resolve(
+		who: &AccountId,
+		credit: CreditOf<AccountId, Self>,
+	) -> Result<(), CreditOf<AccountId, Self>> {
+		let v = credit.peek();
+		let debt = match Self::deposit(who, v) {
+			Err(_) => return Err(credit),
+			Ok(d) => d,
+		};
+		let result = credit.offset(debt).try_drop();
+		debug_assert!(result.is_ok(), "ok deposit return must be equal to credit value; qed");
+		Ok(())
+	}
+
+	/// The balance of `who` is decreased in order to counter `debt`. If the whole of `debt`
+	/// cannot be countered, then nothing is changed and the original `debt` is returned in an
+	/// `Err`.
+	fn settle(
+		who: &AccountId,
+		debt: DebtOf<AccountId, Self>,
+		//TODO: liveness: ExistenceRequirement,
+	) -> Result<CreditOf<AccountId, Self>, DebtOf<AccountId, Self>> {
+		let amount = debt.peek();
+		let credit = match Self::withdraw(who, amount) {
+			Err(_) => return Err(debt),
+			Ok(d) => d,
+		};
+		match credit.offset(debt) {
+			SameOrOther::None => Ok(CreditOf::<AccountId, Self>::zero()),
+			SameOrOther::Same(dust) => Ok(dust),
+			SameOrOther::Other(rest) => {
+				debug_assert!(false, "ok withdraw return must be at least debt value; qed");
+				Err(rest)
+			}
+		}
+	}
+}
+
+/// A fungible token class where the balance can be set arbitrarily.
+///
+/// **WARNING**
+/// Do not use this directly unless you want trouble, since it allows you to alter account balances
+/// without keeping the issuance up to date. It has no safeguards against accidentally creating
+/// token imbalances in your system leading to accidental imflation or deflation. It's really just
+/// for the underlying datatype to implement so the user gets the much safer `Balanced` trait to
+/// use.
+pub trait Unbalanced<AccountId>: Inspect<AccountId> {
```

Comment:
```suggestion
pub trait Unbalanced<AccountId>: Inspect<AccountId> {
```

I might have named them "Tracked" and "Untracked" regarding total issuance, but this seems understandable after reading the comments